packages/graphiql/src/components/GraphiQL.js

```diff
@@ -758,7 +758,29 @@ export class GraphiQL extends React.Component {
 
   handlePrettifyQuery = () => {
     const editor = this.getQueryEditor();
-    editor.setValue(print(parse(editor.getValue())));
+    const editorContent = editor.getValue();
+    const prettifiedEditorContent = print(parse(editorContent));
+
+    if (prettifiedEditorContent !== editorContent) {
+      editor.setValue(prettifiedEditorContent);
+    }
+
+    const variableEditor = this.getVariableEditor();
+    const variableEditorContent = variableEditor.getValue();
+
+    let prettifiedVariableEditorContent;
+
+    try {
+      prettifiedVariableEditorContent = JSON.stringify(
+        JSON.parse(variableEditorContent),
+        null,
+        2
+      );
+    } catch {}
+
+    if (prettifiedVariableEditorContent !== variableEditorContent) {
+      variableEditor.setValue(prettifiedVariableEditorContent);
+    }
```

Comment:
This'll clear variables if they weren't valid; let's not do that. Instead move it all into the `try` block:

```js
try {
  const prettifiedVariableEditorContent = JSON.stringify(JSON.parse(variableEditorContent), null, 2);
  if (prettifiedVariableEditorContent !== variableEditorContent) {
    variableEditor.setValue(prettifiedVariableEditorContent);
  }
} catch {
  /* Parsing JSON failed, skip prettification */
}
```